### DISCUSSION TITLE
Hiding the console ?

### DISCUSSION THREAD

**Discussion by davidovich** [pcottle/learnGitBranching#904] [Q&A]
First, thank you for this excellent project! I have recommended it since I found out about it in 2015 and now I will use it extensively in a training given inside the company.

I am embedding graphs in a reveal-js based presentation (in an `iframe`), and this works very well (I prime the graph by using the `?command=` url query). 

I was wondering if there was some way (I really tried to read all the issues, and the closest I found was #156) to hide or minimize the console to allow for narrower graph presentations. That way, I would have enough space to add a comment side by side in the slide.

Thanks!

**Comment by pcottle** [comment]
Hi David! There's nothing supported in-app to do this; you can do it manually by executing the following javascript in your chrome dev tools:

```
$('#controls').remove()
```

does that work for your use case or do you need something in the URL parameters to prime the removal?

**Reply by davidovich** [reply]
As the console is the primary input point, having it completely disappear would leave us in a bind if we wanted to re-enable the console :-). Maybe a command to rollup the console to limit the vertical place it takes, and upon a click, would roll-back-down. Something very similar to #156.

But I think I could also apply your suggestion (to hide the `#controls`) for the purely graph based tutorials. That would of course not let the participant interact any further with learnGitBranching.

If we had a command, we would get best of both worlds as we would prime `hideConsole` and then the user could re-enable by clicking on the minimized window. 

Or, something like a drawer that shifts left, but I am not enough of a javascript programmer to evaluate if that is too involved.

**Reply by pcottle** [reply]
Ah I see, yeah clicking to minimize is probably the best but it gets tricky to support that in terms of layout (since now the UI has to shift over and pretend the command palette isnt there taking up vertical space). I'll keep this open for now then

**Reply by davidovich** [reply]
Thanks for your responses. Should I open a more formal issue than a discussion?

**Comment by pcottle** [comment]
Sure, so we can track it.

On Mon, Dec 6, 2021 at 2:44 PM David Genest ***@***.***>
wrote:

> Thanks for your responses. Should I open a more formal issue than a
> discussion?
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/pcottle/learnGitBranching/discussions/904#discussioncomment-1760559>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAIVDH5P5FWYJAMTVL4JGHDUPUHCXANCNFSM5JLXECUQ>
> .
>
-- 
-Sent from Gmail Mobile
